new_experiment/nbconverted/viz_expression_data.py

```diff
@@ -0,0 +1,482 @@
+normalized_simulated_data['sample group'] = "simulated"
+
+
+# In[20]:
+
+
+normalized_all_data = pd.concat([normalized_template_data,
+                             normalized_simulated_data,
+                             normalized_compendium_data
+])
+
+
+# In[21]:
+
+
+# Plot
+
+# Drop label column
+normalized_all_data_numeric = normalized_all_data.drop(['sample group'], axis=1)
+
+model = umap.UMAP(random_state=1).fit(normalized_all_data_numeric)
+
+normalized_all_data_UMAPencoded = model.transform(normalized_all_data_numeric)
+normalized_all_data_UMAPencoded_df = pd.DataFrame(data=normalized_all_data_UMAPencoded,
+                                         index=normalized_all_data.index,
+                                         columns=['1','2'])
+
+# Add back label column
+normalized_all_data_UMAPencoded_df['sample group'] = normalized_all_data['sample group']
+
+# Plot
+fig = ggplot(normalized_all_data_UMAPencoded_df, aes(x='1', y='2'))
+fig += geom_point(aes(color='sample group'), alpha=0.1)
+fig += labs(x ='UMAP 1',
+            y = 'UMAP 2',
+            title = 'Gene expression data in gene space')
+fig += theme_bw()
+fig += theme(
+    legend_title_align = "center",
+    plot_background=element_rect(fill='white'),
+    legend_key=element_rect(fill='white', colour='white'), 
+    legend_title=element_text(family='sans-serif', size=15),
+    legend_text=element_text(family='sans-serif', size=12),
+    plot_title=element_text(family='sans-serif', size=15),
+    axis_text=element_text(family='sans-serif', size=12),
+    axis_title=element_text(family='sans-serif', size=15)
+    )
+fig += scale_color_manual(['#bdbdbd', 'red', 'blue'])
+fig += guides(colour=guide_legend(override_aes={'alpha': 1}))
+
+print(fig)
+
+
+# ## Plot gene expression data in latent space
+
+# In[22]:
+
+
+# Model files
+model_encoder_filename = glob.glob(os.path.join(vae_model_dir, "*_encoder_model.h5"))[0]
+weights_encoder_filename = glob.glob(os.path.join(vae_model_dir, "*_encoder_weights.h5"))[0]
+model_decoder_filename = glob.glob(os.path.join(vae_model_dir, "*_decoder_model.h5"))[0]
+weights_decoder_filename = glob.glob(os.path.join(vae_model_dir, "*_decoder_weights.h5"))[0]
+
+
+# In[23]:
+
+
+# Load saved models
+loaded_model = load_model(model_encoder_filename, compile=False)
+loaded_decode_model = load_model(model_decoder_filename, compile=False)
+
+loaded_model.load_weights(weights_encoder_filename)
+loaded_decode_model.load_weights(weights_decoder_filename)
+
+
+# In[24]:
+
+
+# PCA model
+pca = PCA(n_components=2)
+
+
+# In[25]:
+
+
+# Encode compendium
+normalized_compendium = normalized_compendium_data.drop(['sample group'], axis=1)
+compendium_encoded = loaded_model.predict_on_batch(normalized_compendium)
+
+compendium_encoded_df = pd.DataFrame(data=compendium_encoded, 
+                                     index=normalized_compendium.index)
+
+# Get and save PCA model
+model2 = pca.fit(compendium_encoded_df)
```

Comment:
Any reason you're using PCA for the latent space and UMAP for the gene expression space?

Reply:
See comment above: PCA was used to test if the simulation was implemented propoerly -i.e. the simulation produced a linear shift of the template experiment